packages/grid/x-data-grid/src/hooks/utils/useGridApiEventHandler.test.tsx

```diff
@@ -47,14 +47,14 @@ describe('useGridApiEventHandler', () => {
       // which makes 2 event listeners to be registered. Since the second render is never
       // committed (to simulate a trashed render in React 18), the effects also don't run, so we're
       // unable to unsubscribe the last listener using the cleanup function.
-      expect(apiRef.current.subscribeEvent.callCount).to.equal(2);
+      expect(apiRef.current.subscribeEvent.callCount).to.equal(3);
```

Comment:
I'm not sure I understand why it changes from 2 to 3. Can you briefly explain it?

Reply:
This test checks the number of calls to `apiRef.current.subscribeEvent`. Inside this hook there're 2 places where this method is called:

Synchronously: https://github.com/mui/mui-x/blob/9af5f06db0b617625f4c616d7c16af1d70e37d3f/packages/grid/x-data-grid/src/hooks/utils/useGridApiEventHandler.ts#L56

Inside the effect: https://github.com/mui/mui-x/blob/9af5f06db0b617625f4c616d7c16af1d70e37d3f/packages/grid/x-data-grid/src/hooks/utils/useGridApiEventHandler.ts#L88

In React 17, it's called synchronously first + effect = 2
In React 18, it's called synchronously first + effect + effect again because of StrictEffects = 3

StrictEffect doesn't care about the dependencies. It calls the effect again.